Shared/Samples/Identify layer features/IdentifyLayerFeaturesView.swift

```diff
@@ -0,0 +1,182 @@
+            let map = Map(basemapStyle: .arcGISTopographic)
+            
+            // Center the map on the USA.
+            map.initialViewpoint = Viewpoint(
+                center: Point(x: -10977012.785807, y: 4514257.550369, spatialReference: .webMercator),
+                scale: 68015210
+            )
+            return map
+        }()
+        
+        /// A Boolean value indicating whether to show a layer alert.
+        @Published var isShowingLayerAlert = false
+        
+        /// The title for the identify layer result alert.
+        @Published var layerAlertTitle = ""
+        
+        /// The message for the identify layer result alert.
+        @Published var layerAlertMessage = ""
+        
+        /// A Boolean value indicating whether to show an error alert.
+        @Published var isShowingErrorAlert = false
+        
+        /// The error shown in the error alert.
+        @Published var error: Error? {
+            didSet { isShowingErrorAlert = error != nil }
+        }
+        
+        /// Update the identify layer result alert text based on the identify layer results.
+        /// - Parameter results: An `IdentifyLayerResult Array` to handle.
+        func handleIdentifyResults(_ results: [IdentifyLayerResult]) {
+            var alertMessageString = ""
+            var totalGeoElementsCount = 0
+            
+            for (iCount, identifyLayerResult) in results.enumerated() {
+                // Create alert message the from geoElement count and the layer name.
+                let geoElementsCount = geoElementsCountFromResult(identifyLayerResult)
+                let layerName = identifyLayerResult.layerContent.name
+                alertMessageString.append("\(layerName): \(geoElementsCount)")
+                
+                // Add new line character if not the final element in array.
+                if iCount != results.count - 1 {
+                    alertMessageString.append("\n")
+                }
+                
+                // Update total count.
+                totalGeoElementsCount += geoElementsCount
+            }
+            
+            if totalGeoElementsCount > 0 {
+                // Show the results if any elements were found.
+                layerAlertTitle = "Number of elements found"
+                layerAlertMessage = alertMessageString
+            } else {
+                // Notify user that no elements were found.
+                layerAlertTitle = "No element found"
+                layerAlertMessage = ""
+            }
+        }
+        
+        /// Count the geoElements from an identify layer result.
+        /// - Parameter result: The `IdentifyLayerResult` to count.
+        /// - Returns: An `Int` count of the geoElements
+        private func geoElementsCountFromResult(_ result: IdentifyLayerResult) -> Int {
+            var tempResults = [result]
+            
+            // Using Depth First Search approach to handle recursion.
```

Comment:
As far as I can tell, there is no DFS in this method. It is more of a BFS, but I would like to refactor it to be more efficient - no need to hold a whole array of temp results while the only thing we need is an integer count.

Reply:
I figured it out using recursion. Be proud.